docs/decisions/0008-zero-state-browsing-with-universal-link.rst

```diff
@@ -0,0 +1,164 @@
+to direct the user to the correct enterprise means that the link would still work if an enterprise changes its slug.
+
+A user will be prompted to logistrate when visiting the link if not already logged in and be redirected to the learner portal afterwards.
+Currently, a not found page is shown on the learner portal if the user is not linked. If the user is not linked and an ``enterprise_customer_key`` is found in the query params,
+the portal will call the new endpoint to link the user. If the call succeeds, the portal will reload and the user will be able to browse as usual.
+If the call fails, appropriate messaging regarding the failure will be shown to the user. 
+This will also link users that were already logged in before visiting the link.
+
+Limiting logistration through universal link
+--------------------------------------------
+
+Any user with a universal link that contains a valid key is able to link themselves with an enterprise. We want to take necessary precautions to prevent abuse of this feature.
+`EnterpriseCustomerKey` will be expirable and have a limit on the number of times it can be used. We will validate the key (not expired and usage limit has not been reached)
+when linking a user to an enterprise. If the key is invalid, the user will not be linked and be informed that the link is invalid. We will also validate that
+the ``enterprise_customer_uuid`` in the request payload matches with the ``enterprise_customer_uuid`` associated with the key. 
+
+Limiting number of course requests to prevent email spamming
+------------------------------------------------------------
+
+In the future we want to limit the number of course requests to prevent admins from being spammed with request notifications.
+This should be factored in when implementing course request and will not be addressed here.
+
+Expiring a universal link
+-------------------------
+
+A universal link can have a lifetime dictated by the expiry date set on the ``EnterpriseCustomerKey`` used in the link. A key will not be
+valid if expired. An enterprise admin should be able to manually expire a key to revoke a link.
+
+
+Analytics
+---------
+
+Segment events will be fired when
+* an ``enterprise_customer_key`` is created - i.e. 'edx.bi.user.enterprise.key.created'
+* a user is linked to an enterprise using the universal link, include key used in the event - 'edx.bi.user.enterprise.key.used'
+* a key expires or reaches its usage cap - 'edx.bi.user.enterprise.key.invalidated'
+* a user attempts to use a bad key - 'edx.bi.user.enterprise.key.attempted'
+
+We will also keep track of which user were linked using a universal link in the model (possibily a new field on ``EnterpriseCustomerUser``) and include that information
+in future tracking events, i.e. search events in the learner portal. 
+
+We also want to track how many users land on the logistration page but abandons the page. This could possibly be done through google analytics.
```

Comment:
I think a clicked event would also be good providing the user ID, key_UUID and enterprise_UUID

Reply:
could we get this information from the page events we currently have?

Reply:
Perhaps? I could just filter the page views on the presence of `enterprise_customer_key` in the URL?